projects/kit/styles/components/progress-bar.less

```diff
@@ -104,6 +104,18 @@
         border-radius: inherit;
     }
 
+    &[tuiProgressFixedGradient] {
```

Comment:
Let's not augment this file with this logic.
In far future the basic version of `Progress` component becomes framework-agnostic component and it will know nothing about this directive.

Let create folder `fixed-gradient` inside `projects/kit/components/progress/progress-bar`:
```
├── fixed-gradient
│   ├── progress-fixed-gradient.directive.ts
│   └── progress-fixed-gradient.style.less
├── progress-bar.component.ts
└── progress-color-segments.directive.ts
```

Put all new styles inside `progress-fixed-gradient.style.less`:
```less
.fixed-gradient() {
    inline-size: 100% !important;
    clip-path: inset(0 calc(100% - var(--tui-progress-percent)) 0 0 round var(--tui-radius-m));
    transition: clip-path var(--tui-duration) linear;
}

&[tuiProgressFixedGradient] {
    &::-moz-progress-bar {
        .fixed-gradient();

        margin-right: calc(-100% + var(--tui-progress-percent));
    }

    &::-webkit-progress-value {
        .fixed-gradient();
    }
}
```


And  file `progress-fixed-gradient.directive.ts` contains:

```ts
@Component({
    standalone: true,
    template: '',
    styleUrls: ['./progress-fixed-gradient.style.less'],
    encapsulation: ViewEncapsulation.None,
    changeDetection: ChangeDetectionStrategy.OnPush,
    host: {
        class: 'tui-fixed-gradient',
    },
})
class TuiProgressFixedGradientStyles {}

@Directive({
    standalone: true,
    selector: 'progress[tuiProgressBar][tuiProgressFixedGradient]',
    // [...]
})
export class TuiProgressFixedGradientDirective {
    protected readonly nothing = tuiWithStyles(TuiProgressFixedGradientStyles);

    // [...]
}
```

Reply:
@taiga-family/core-team Guys, what do you think about my last suggestion ?
Is it not overkill for this case ?